bson/json_util.py

```diff
@@ -311,6 +311,35 @@ def _arguments_repr(self):
                     self.json_mode,
                     super(JSONOptions, self)._arguments_repr()))
 
+    def with_options(self, **kwargs):
+        """
+        Make a copy of this JSONOptions, overriding some options::
+
+            >>> from bson.json_util import CANONICAL_JSON_OPTIONS
+            >>> CANONICAL_JSON_OPTIONS.tz_aware
+            True
+            >>> json_options = CANONICAL_JSON_OPTIONS.with_options(tz_aware=False)
+            >>> json_options.tz_aware
+            False
+
+        .. versionadded:: 3.12
+        """
+        opts = ('strict_number_long',
+                'datetime_representation',
+                'strict_uuid',
+                'json_mode',
+                'document_class',
+                'tz_aware',
+                'uuid_representation',
+                'unicode_decode_error_handler',
+                'tzinfo',
+                'type_registry')
```

Comment:
It might be more future proof to use `self._fields` to get all the fields from CodecOptions. https://docs.python.org/3/library/collections.html#collections.somenamedtuple._fields

Reply:
I used `_as_dict()` instead as it seemed more tailored to our usecase.